sklearn/svm/src/liblinear/liblinear_helper.c

```diff
@@ -38,12 +40,21 @@ static struct feature_node **dense_to_sparse(double *x, int n_samples,
         sparse[i] = T;
 
         for (j=1; j<=n_features; ++j) {
-            if (*x != 0) {
-                T->value = *x;
-                T->index = j;
-                ++ T;
+            if (double_precision) {
```

Comment:
Is the optimiser likely to compile this out of the loop?

Reply:
Yes, but even if it doesn't, the CPU's branch predictor will reduce the cost of the if statement to zero.